refgenconf/refgenconf.py

```diff
@@ -140,6 +148,127 @@ def list_genomes_by_asset(self, asset=None):
         return self._invert_genomes() \
             if not asset else [g for g, am in self.genomes.items() if asset in am]
 
+    def list_remote(self, get_url=lambda rgc: "{}/assets".format(rgc.genome_server)):
+        """
+        List genomes and assets available remotely.
+
+        :param function(refgenconf.RefGenConf) -> str get_url: how to determine
+            URL request, given RefGenConf instance
+        :return str, str: text reps of remotely available genomes and assets
+        """
+        url = get_url(self)
+        _LOGGER.info("Querying available assets from server: {}".format(url))
+        genomes, assets = _list_remote(url)
+        return genomes, assets
+
+    def pull_asset(self, genome, assets, genome_config, unpack=True,
+                   get_json_url=lambda base, g, a: "{}/asset/{}/{}".format(base, g, a),
+                   get_main_url=None):
+        """
+        Download and possibly unpack one or more assets for a given ref gen.
+
+        :param str genome: name of a reference genome assembly of interest
+        :param str assets: name(s) of particular asset(s) to fetch
+        :param str genome_config: path to genome configuration file to update
+        :param bool unpack: whether to unpack a tarball
+        :param function(str, str, str) -> str get_json_url: how to build URL from
+            genome server URL base, genome, and asset
+        :param function(str) -> str get_main_url: how to get archive URL from
+            main URL
+        :return Iterable[(str, str | NoneType)]: collection of pairs of asset
+            name and folder name (key-value pair with which genome config file
+            is updated) if pull succeeds, else asset key and a null value.
+        """
+        if isinstance(assets, str):
+            assets = [assets]
+        elif not isinstance(assets, Iterable):
+            raise TypeError("Assets to pull should be single name or collection "
+                            "of names; got {} ({})".format(assets, type(assets)))
+        return [self._pull_asset(
+            genome, a, genome_config, unpack, get_json_url, get_main_url) for a in assets]
+
+    def _pull_asset(self, genome, asset, genome_config, unpack, get_json_url, get_main_url):
+
+        _LOGGER.info("Starting pull for {}: {}".format(genome, asset))
+
+        def raise_unpack_error():
+            raise NotImplementedError("Tarball preservation isn't yet supported.")
```

Comment:
isn't this message a bit misleading? Currently we leave the tarballs in place after "untaring", so they are preserved in a way

Reply:
@michalstolarczyk can you change it to how you would suggest?